src/commands/init.js

```diff
@@ -83,84 +222,323 @@ Examples:
       return
     }
 
-    let installCommand = yarn ? 'yarn' : 'npm install'
-    let codegenCommand = yarn ? 'yarn codegen' : 'npm run codegen'
-    let deployCommand = yarn ? 'yarn deploy' : 'npm run deploy'
+    let commands = {
+      install: yarn ? 'yarn' : 'npm install',
+      codegen: yarn ? 'yarn codegen' : 'npm run codegen',
+      deploy: yarn ? 'yarn deploy' : 'npm run deploy',
+    }
 
-    // Clone the example subgraph repository
-    let spinner
-    try {
-      spinner = print.spin('Cloning example subgraph')
-      await system.run(
-        `git clone http://github.com/graphprotocol/example-subgraph ${directory}`
+    // If all parameters are provided from the command-line,
+    // go straight to creating the subgraph from the example
+    if (fromExample && subgraphName && directory) {
+      return await initSubgraphFromExample(
+        toolbox,
+        { allowSimpleName, directory, subgraphName },
+        { commands }
       )
-      spinner.stop()
-    } catch (e) {
-      spinner.fail(`Failed to clone example subgraph repository: ${e}`)
-      process.exitCode = 1
-      return
     }
 
-    // Update package.json to match the subgraph name
-    try {
-      // Load package.json
-      let pkgJsonFilename = filesystem.path(directory, 'package.json')
-      let pkgJson = await filesystem.read(pkgJsonFilename, 'json')
+    // If all parameters are provided from the command-line,
+    // go straight to creating the subgraph from an existing contract
+    if (fromContract && subgraphName && directory && network) {
+      let abi
+      try {
+        abi = await loadAbiFromEtherscan(network, fromContract)
+      } catch (e) {
+        process.exitCode = 1
```

Comment:
should we not print `e` here?

Reply:
It's not obvious, but since `loadAbiFromEtherscan` uses a spinner, the spinner takes care of printing an error when it fails.